### PR TITLE
New mod "taskbar-disappearing-icon-fix-win11"

### PR DESCRIPTION
**Pull request by yezhiyi9670**
Try to fix the annoying disappearing taskbar icons bug in Windows 11 while switching virtual desktops.

### CODE REVIEW

File: mods/taskbar-disappearing-icon-fix-win11.wh.cpp [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+            {
+                LR"(public: void __cdecl winrt::Taskbar::implementation::TaskListButton::DisplayName(struct winrt::hstring))",
+                LR"(public: void __cdecl winrt::Taskbar::implementation::TaskListButton::DisplayName(struct winrt::hstring) __ptr64)",
+            },
+            (void**)&TaskListButton_Hook2_Original,
+            (void*)TaskListButton_Hook2_Hook,
+        },
+    };
+
+    return HookSymbolsWithOnlineCacheFallback(module, symbolHooks,
```
Comment (m417z) [inline mods/taskbar-disappearing-icon-fix-win11.wh.cpp]: Please `#include <windhawk_utils.h>` and use `WindhawkUtils::HookSymbols` for this.

**Review by m417z — COMMENTED** [review]
Thanks! It's something I wanted to try and fix as well, but didn't get the chance to get to it

File: mods/taskbar-disappearing-icon-fix-win11.wh.cpp [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+/*
+# Taskbar Disappearing Icon Fix for Windows 11
+
+Tries to fix disappearing icon bug in Windows 11 taskbar while switching virtual desktops.
+
+Since I cannot reproduce this bug very reliably, this mod may not have fixed the bug entirely. Let me know if it still persists.
+
+Takeaway message:
+
+- It is observed that hiding (`Visibility: Collpased`) the invisible icon and then making it visible after a while will fix it.
```
Comment (m417z) [inline mods/taskbar-disappearing-icon-fix-win11.wh.cpp]: Typo at "Collpased".

File: mods/taskbar-disappearing-icon-fix-win11.wh.cpp [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+
+    return HookSymbolsWithOnlineCacheFallback(module, symbolHooks,
+                                              ARRAYSIZE(symbolHooks));
+}
+
+BOOL ModInitWithTaskbarView(HMODULE taskbarViewModule) {
+    if (!HookTaskbarViewDllSymbols(taskbarViewModule)) {
+        return FALSE;
+    }
+    if (!HookTaskbarDllSymbols()) {
```
Comment (m417z) [inline mods/taskbar-disappearing-icon-fix-win11.wh.cpp]: Remove this call.

File: mods/taskbar-disappearing-icon-fix-win11.wh.cpp [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+    auto taskListButtonElement = taskListButtonIUnknown.as<FrameworkElement>();
+
+    HideAndShowIcon(taskListButtonElement);
+}
+
+// ===========================================================
+
+bool HookTaskbarViewDllSymbols(HMODULE module) {
+    // taskbar.view.dll
```
Comment (m417z) [inline mods/taskbar-disappearing-icon-fix-win11.wh.cpp]:
```suggestion
    // taskbar.view.dll, ExplorerExtensions.dll
```

Comment (yezhiyi9670) [reply mods/taskbar-disappearing-icon-fix-win11.wh.cpp]: I am just wondering why ExplorerExtensions.dll is also considered a target. The symbols I use are found in taskbar.view.dll.

Comment (m417z) [reply mods/taskbar-disappearing-icon-fix-win11.wh.cpp]: Windows 11 version 21H2 had ExplorerExtensions.dll instead of taskbar.view.dll.

**Comment by Marshall-Brooks** [comment]
Just wanted to comment that I was having an intermittent issue with disappearing taskbar icons (not with virtual desktops) and I haven't seen the issue since installing the mod, so it seems to be working - although the issue was intermittent, so hard to know for sure that it is fixed.

Thank you for developing it and notifying me about it!

**Comment by usernamelectro** [comment]
Unfortunately, it happened again and the icons were missing after I woke my PC this evening.

screenshot

**Comment by Marshall-Brooks** [comment]
Need to update - it isn't perfectly working for me either. I am using both your mod and "Virtual Desktop Preserve Taskbar Order". I thought your mod had resolved the issue, so I disabled the preserve Taskbar Order mod and the issue re-appeared.

It's difficult to troubleshoot b/c for me the issue ONLY appears with Microsoft Access and ONLY sometimes when I have two or more databases open.

**Comment by yezhiyi9670** [comment]
@usernamelectro @Marshall-Brooks I need more specific info to try to make a fix, including:

- System version
- Detailed description: Is the mod loaded successfully (check the Loaded Mods list)? Does the icon reappear if you click on it? Does the icon reappear a few seconds later even if you do nothing? Is it related to virtual desktops? How to reproduce (if you can find a way)?

Please open an issue to provide your information and track the issue.

**Pull request merged**